javascript/ql/src/RegExp/IdentityReplacement.ql

```diff
@@ -0,0 +1,44 @@
+/**
+ * @name Replacement of a substring with itself
+ * @description Replacing a substring with itself has no effect and may indicate a mistake.
+ * @kind problem
+ * @problem.severity warning
+ * @id js/identity-replacement
+ * @tags correctness
+ * @precision very-high
+ */
+
+import javascript
+
+/**
+ * Holds if `e`, when used as the first argument of `String.prototype.replace`, matches
+ * `s` and nothing else.
+ */
+predicate matchesString(Expr e, string s) {
+  exists (RegExpConstant c |
+    matchesConstant(e.(RegExpLiteral).getRoot(), c) and
```

Comment:
We should exclude the `i`-flag for cases like: `x.replace(/foo/i, 'foo')`.

Reply:
An excellent point, thanks!